truffle/src/main/java/org/jruby/truffle/nodes/core/ExceptionNodes.java

```diff
@@ -114,6 +115,7 @@ public BacktraceNode(RubyContext context, SourceSection sourceSection) {
             readCustomBacktrace = new ReadHeadObjectFieldNode("@custom_backtrace");
         }
 
+        @TruffleBoundary
```

Comment:
You're right, the tool shows that asRubyStringArray is the method that makes a lot reachable for compilation.